File: day1/increments.cs
class increments {
    public static void Main(string[] args) {
        int[] win = new int[3]{0,0,0};
        int prv = 0;
        int prw = 0;
        int cnv = 0;
        int cnw = 0;
        var lines = System.IO.File.ReadLines("input");
        int idx = 0;
        foreach (var line in lines) {
            int val = System.Convert.ToInt32(line);
            // populate the window
            win[idx%3] = val;
            // compare (if beyond first value)
            if (idx>0) {
                if (val>prv) {
                    cnv = cnv+1;
                }
            }
            // calculate sum
            int sum = win[0]+win[1]+win[2];
            // compare window (if beyond first three values)
            if (idx>2) {
                if (sum>prw) {
                    cnw = cnw+1;
                }
            }
            prv = val;
            prw = sum;
            idx = idx+1;
        }
        System.Console.WriteLine($"Increments: single:{cnv} windowed:{cnw}");
    }
}

File: day1/increments.rs
use std::fs::File;
use std::io::{BufRead,BufReader};

fn main() {
    let file = File::open("input").unwrap();
    let rdr = BufReader::new(file);
    let mut win : [i32; 3] = [0, 0, 0];
    let mut prv : i32 = 0;
    let mut prw : i32 = 0;
    let mut cnv : i32 = 0;
    let mut cnw : i32 = 0;
    for (idx,line) in rdr.lines().enumerate() {
        let val : i32 = line.unwrap().parse().unwrap();
        // populate the window
        win[idx%3] = val;
        // compare (if beyond first value)
        if idx>0 {
            if val>prv {
                cnv = cnv+1;
            }
        }
        // calculate sum
        let sum : i32 = win[0]+win[1]+win[2];
        // compare window (if beyond first three values)
        if idx>2 {
            if sum>prw {
                cnw = cnw+1
            }
        }
        prv = val;
        prw = sum;
    }
    println!("Increments: single:{} windowed:{}", cnv, cnw);
}
